internal/cmd/controller/agentmanagement/controllers/clusterregistrationtoken/handler.go

```diff
@@ -69,7 +69,7 @@ func Register(ctx context.Context,
 
 func (h *handler) OnChange(token *fleet.ClusterRegistrationToken, status fleet.ClusterRegistrationTokenStatus) ([]runtime.Object, fleet.ClusterRegistrationTokenStatus, error) {
 	if gone, err := h.deleteExpired(token); err != nil || gone {
-		return nil, status, nil
+		return nil, status, err
```

Comment:
I wonder if this may have unforeseen consequences :thinking:

Reply:
I was not sure, but better to keep the previous behavior then to end up with retry loops.